qubrabench/utils/plotting_strategy.py

```diff
@@ -0,0 +1,185 @@
+import math
+from abc import ABC, abstractmethod
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import matplotlib.colors as mcolors
+
+
+class PlottingStrategy(ABC):
+    """
+    Generic plotting strategy to visualize benchmarking data.
+
+    Assumes data is given as a pandas dataframe.
+    See the abstract methods below for configuration options.
+    """
+
+    # colors for each dataset in the plot
+    colors: dict[str, str] = {}
+
+    @abstractmethod
+    def x_axis_column(self) -> str:
+        """
+        Data column to plot along the X-axis
+        """
+        return ""
+
+    @abstractmethod
+    def x_axis_label(self) -> str:
+        """
+        Label to display for the X-axis
+        """
+        return ""
+
+    @abstractmethod
+    def y_axis_label(self) -> str:
+        """
+        Label to display for the Y-axis
+        """
+        return ""
+
+    @abstractmethod
+    def columns_to_group_for_plots(self) -> list[str]:
```

Comment:
```suggestion
    def get_plot_column_group_labels(self) -> list[str]:
```
Since these are essentially getters I suggest this phrasing. The docstring and usage details what is supposed to happen very clearly imo.

Also remove the TODO and update references